### PR TITLE
Chunk metadata and fix cluster data model

### PR DESCRIPTION
**Pull request by knapii-developments**
This PR:

-  Addresses typos made to field values in cluster data model ([SCP-1866](https://broadworkbench.atlassian.net/browse/SCP-1866))
- A solution, chunking,  for documents that are over FireStore document limits for cell metadata files ([SCP-1871](https://broadworkbench.atlassian.net/browse/SCP-1871))
- Adds [unit test](https://github.com/broadinstitute/scp-ingest-pipeline/pull/25/commits/378d89621e3218c16cb47f1dbfc7805039246fad#diff-78315c5177969687c6f0ee9c557aa770) for chunking cell metadata subdocuments
- Adds self.unique_values for an annotation value in the cell metadata model. 
- Populates the top-level cluster document attribute for "[cell_annotations](https://github.com/broadinstitute/scp-ingest-pipeline/blob/378d89621e3218c16cb47f1dbfc7805039246fad/ingest/clusters.py#L50)" ([SCP-1866](https://broadworkbench.atlassian.net/browse/SCP-1866))
- Case insensitive search in headers to determine if cluster type is 3D or not ([SCP-1866](https://broadworkbench.atlassian.net/browse/SCP-1866))

### CODE REVIEW

File: tests/data/10k_cells_29k_genes.cluster.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-NAME	X	Y	Z
+NAME	X	Y	z
```
[the remark this answers is not in the document]
Comment (bistline) [reply tests/data/10k_cells_29k_genes.cluster.txt]: Agreed - this way we have coverage for that corner case.

File: tests/test_cell_metadata.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            'doc_name', 'fake/path/cell_metadata', 'ClusterID'
+        ):
+            pass
+
+        sys.stdout = sys.__stdout__
+        # String represents outputs from chunk_subdocuments method
+        #  number of bytes, end_index, starting index, number of bytes, end_index, starting index
+        expected = "1048562 , 17740, 0 , 17739\n347983 , 23593, 17740 , 23593\n"
+
+        self.assertEqual(expected, captured_output.getvalue())
```
[the remark this answers is not in the document]
Comment (bistline) [reply tests/test_cell_metadata.py]: I'll agree with Eric on this - we should start using more robust logging practices, especially since we're going to want to write this to Stackdriver eventually.

[the remark this answers is not in the document]
Comment (knapii-developments) [reply tests/test_cell_metadata.py]: I agree. This issue is applicable to other use cases, for example validation errors for cell metadata. I think one option to consider is writing these statements to [in memory file or temp files](https://docs.pytest.org/en/2.8.7/capture.html). For now I'll make a comment.
